src/legacy/core_plugins/kibana/public/visualize/embeddable/visualize_embeddable_factory.tsx

```diff
@@ -81,6 +81,13 @@ export class VisualizeEmbeddableFactory extends EmbeddableFactory<
   private visTypes?: VisTypesRegistry;
   public readonly type = VISUALIZE_EMBEDDABLE_TYPE;
 
+  static async getInstance(): Promise<VisualizeEmbeddableFactory> {
+    const instance = new VisualizeEmbeddableFactory();
+    await instance.initializeVisTypes();
+
+    return instance;
+  }
+
```

Comment:
(suggestion) What if we make `VisualizeEmbeddableFactory` take `visTypes` as a parameter in the constructor?  Then no risk of someone using `VisualizeEmbeddableFactory` elsewhere and instantiating a new one without using `getInstance`?
```
  static async createVisualizeEmbeddableFactory(): Promise<VisualizeEmbeddableFactory> {
    const $injector = await chrome.dangerouslyGetActiveInjector();
    const Private = $injector.get<IPrivate>('Private');
    const visTypes = Private(VisTypesRegistryProvider);

    return new VisualizeEmbeddableFactory(visTypes);
  }
```

Reply:
looks like a good idea. Thanks